src/core/thread/thread_netif.cpp

```diff
@@ -49,7 +49,7 @@ void ThreadNetif::Up(void)
 
     // Enable the MAC just in case it was disabled while the Interface was down.
     Get<Mac::Mac>().SetEnabled(true);
-#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE && OPENTHREAD_CONFIG_CHANNEL_MONITOR_AUTO_START_ENABLE
     IgnoreError(Get<Utils::ChannelMonitor>().Start());
 #endif
```

Comment:
We should be consistent with `OPENTHREAD_CONFIG_CHANNEL_MONITOR_AUTO_START_ENABLE`. If it is supposed to control automatically starting the feature from `ThreadNetif::Up()`, it makes sense to also control it being automatically stopped in `ThreadNetif::Down()`.

It is either automatically started and stopped by the OpenThread stack itself, or it must be controlled by the next layer (application) code. We cannot delegate starting to the next layer but allow it to be automatically stopped.